servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/TcpServerBinder.java

```diff
@@ -169,6 +205,116 @@ protected void handleSubscribe(Subscriber<? super ServerContext> subscriber) {
+            final ReadOnlyTcpServerConfig config,
+            @Nullable final EarlyConnectionAcceptor earlyConnectionAcceptor,
+            @Nullable final LateConnectionAcceptor lateConnectionAcceptor,
+            @Nullable final InfluencerConnectionAcceptor connectionAcceptor) {
+        final Executor offloadExecutor = executionContext.executor();
+
+        if (earlyConnectionAcceptor != null) {
+            final ExecutionContext<?> channelExecutionContext = channelExecutionContext(channel, executionContext);
+
+            ConnectionInfo info = new ConnectionInfo() {
+                @Override
+                public SocketAddress localAddress() {
+                    return channel.localAddress();
+                }
+
+                @Override
+                public SocketAddress remoteAddress() {
+                    return channel.remoteAddress();
+                }
+
+                @Override
+                public ExecutionContext<?> executionContext() {
+                    return channelExecutionContext;
+                }
+
+                @Nullable
+                @Override
+                public SslConfig sslConfig() {
+                    return config.sslConfig();
+                }
+
+                @Nullable
+                @Override
+                public SSLSession sslSession() {
+                    return null;
+                }
+
+                @Nullable
+                @Override
+                public <T> T socketOption(final SocketOption<T> option) {
+                    return getOption(option, channel.config(), config.idleTimeoutMs());
+                }
+
+                @Override
+                public Protocol protocol() {
+                    return () -> "TCP";
+                }
+            };
+
+            final EarlyConnectionAcceptorHandler acceptorHandler = new EarlyConnectionAcceptorHandler();
+            channel.pipeline().addLast(acceptorHandler);
+
+            // Defer is required to isolate the context between the user accept callback and the rest of the connection
+            Completable earlyCompletable = Completable.defer(() -> earlyConnectionAcceptor.accept(info));
+
+            if (earlyConnectionAcceptor.requiredOffloads().isConnectOffloaded()) {
+                earlyCompletable = earlyCompletable.subscribeOn(offloadExecutor);
+            }
+
+            connection = earlyCompletable
+                    .publishOn(channelExecutionContext.ioExecutor(), () -> !channel.eventLoop().inEventLoop())
+                    .concat(connection)
+                    .whenFinally(acceptorHandler::releaseEvents);
+        }
+
+        if (lateConnectionAcceptor != null) {
+            connection = connection.flatMap(conn -> {
+                // Defer is required to isolate the context between the user accept callback and
+                // the rest of the connection
+                Single<CC> deferred = defer(() -> lateConnectionAcceptor.accept(conn).concat(succeeded(conn)));
```

Comment:
I've looked into it and I'm not sure if this will work as you described since we need the `ConnectionContext` as an argument on the `accept` - so we need to stay inside the `flatMap` - maybe I am missing something?